src/main/java/org/sonarsource/sonarlint/ls/AnalysisManager.java

```diff
@@ -456,4 +486,83 @@ private void analyzeAllUnboundOpenFiles() {
     }
   }
 
+  private void analyzeAllOpenJavaFiles() {
```

Comment:
I'm wondering whether it would make sense to refactor these `analyzeAllXxx` methods, e.g with a method named `analyzeFiles(Predicate<URI>)`.

Reply:
I think using predicates will make it harder to understand what this is doing. Just to save the loop, I don't think it worth factorizing. WDYT?